src/Sarif.FunctionalTests/PropertyBagConverterTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Microsoft.CodeAnalysis.Sarif.Readers;
+using Microsoft.CodeAnalysis.Sarif.TestUtilities;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.FunctionalTests
+{
+    public class PropertyBagConverterTests
+    {
+        [Fact]
+        [Trait(TestTraits.Bug, "1045")]
+        public void PropertyBagConverter_RoundTripsStringPropertyWithEscapedCharacters()
+        {
+            string originalLog =
+@"{
+  ""version"": """ + SarifUtilities.SarifFormatVersion + @""",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": ""CodeScanner""
+      },
+      ""properties"": {
+        ""int"": 42,
+        ""string"": ""'\""\\'""
+      }
```

Comment:
@boAndron: This is a minimal repro based on the sample you provided. It fails without the fix, exactly as your sample did, and succeeds with the fix.